src/UnitTransformedLikelihood.cc

```diff
@@ -108,20 +108,25 @@ void setupArrays(
     ndarray::Array<Pixel,1,1> const & data,
     ndarray::Array<Pixel,1,1> const & variance,
     ndarray::Array<Pixel,1,1> const & weights,
-    ndarray::Array<Pixel,1,1> & unweightedData,
+    ndarray::Array<Pixel,1,1> const & unweightedData,
     bool usePixelWeights
 ) {
     afw::detection::flattenArray(footprint, image.getImage()->getArray(), data, image.getXY0());
     afw::detection::flattenArray(footprint, image.getVariance()->getArray(), variance, image.getXY0());
-    if (usePixelWeights) {
-        unweightedData = ndarray::copy(data);
-        // Convert from variance to weights (1/sigma); this is actually the usual inverse-variance
-        // weighting, because we implicitly square it later.
-        weights.asEigen<Eigen::ArrayXpr>() = variance.asEigen<Eigen::ArrayXpr>().sqrt().inverse();
-        data.asEigen<Eigen::ArrayXpr>() *= weights.asEigen<Eigen::ArrayXpr>();
-    } else {
-        unweightedData = data;
+    unweightedData.deep() = data;
+    // Convert from variance to weights (1/sigma); this is actually the usual inverse-variance
+    // weighting, because we implicitly square it later.
+    weights.asEigen<Eigen::ArrayXpr>() = variance.asEigen<Eigen::ArrayXpr>().sqrt().inverse();
```

Comment:
You're taking a `sqrt` here just to square it later (or take a log); isn't that a bit wasteful? Maybe it's not important because the model evaluations consume far more time?